src/Illuminate/Http/Concerns/InteractsWithInput.php

```diff
@@ -375,6 +375,40 @@ public function file($key = null, $default = null)
         return data_get($this->allFiles(), $key, $default);
     }
 
+    /**
+     * Check the list of expected keys, and return the keys among them that are
+     * present in the request.
+     *
+     * @param array $expectedKeys
+     * @return array
+     */
+    public function presentKeys(array $expectedKeys): array
+    {
+        $presentKeys = [];
+
+        foreach ($expectedKeys as $expectedKey) {
+            if ($this->has($expectedKey)) {
+                $presentKeys[] = $expectedKey;
+            }
+        }
+
+        return $presentKeys;
+    }
+
+    /**
+     * Check the list of expected keys, and return the values of the keys that
+     * are present in the request.
+     *
+     * @param array $expectedKeys
+     * @return array
+     */
+    public function valuesOfPresentKeys(array $expectedKeys): array
```

Comment:
Remove the return type here.

```suggestion
    public function valuesOfPresentKeys(array $expectedKeys)
```